x/marketmaker/keeper/proposal_handler.go

```diff
@@ -0,0 +1,197 @@
+			sdk.NewEvent(
+				types.EventTypeIncludeMarketMaker,
+				sdk.NewAttribute(types.AttributeKeyAddress, p.Address),
+				sdk.NewAttribute(types.AttributeKeyPairId, fmt.Sprintf("%d", p.PairId)),
+			),
+		})
+	}
+	return nil
+}
+
+// ExclusionMarketMakers is a handler for exclude eligible market makers.
+func (k Keeper) ExclusionMarketMakers(ctx sdk.Context, proposals []types.MarketMakerHandle) error {
+	for _, p := range proposals {
+		mmAddr, err := sdk.AccAddressFromBech32(p.Address)
+		if err != nil {
+			return err
+		}
+		mm, found := k.GetMarketMaker(ctx, mmAddr, p.PairId)
+		if !found {
+			return sdkerrors.Wrapf(types.ErrNotExistMarketMaker, "%s is not market maker", p.Address)
+		}
+
+		if !mm.Eligible {
+			return sdkerrors.Wrapf(types.ErrInvalidExclusion, "%s is not eligible market maker", p.Address)
+		}
+
+		k.DeleteMarketMaker(ctx, mmAddr, p.PairId)
+
+		ctx.EventManager().EmitEvents(sdk.Events{
+			sdk.NewEvent(
+				types.EventTypeExcludeMarketMaker,
+				sdk.NewAttribute(types.AttributeKeyAddress, p.Address),
+				sdk.NewAttribute(types.AttributeKeyPairId, fmt.Sprintf("%d", p.PairId)),
+			),
+		})
+	}
+	return nil
+}
+
+// RejectionMarketMakers is a handler for reject applied and not eligible market makers.
+func (k Keeper) RejectionMarketMakers(ctx sdk.Context, proposals []types.MarketMakerHandle) error {
+	for _, p := range proposals {
+		mmAddr, err := sdk.AccAddressFromBech32(p.Address)
+		if err != nil {
+			return err
+		}
+
+		mm, found := k.GetMarketMaker(ctx, mmAddr, p.PairId)
+		if !found {
+			return sdkerrors.Wrapf(types.ErrNotExistMarketMaker, "%s is not market maker", p.Address)
+		}
+
+		if mm.Eligible {
+			return sdkerrors.Wrapf(types.ErrInvalidRejection, "%s is eligible market maker", p.Address)
+		}
+
+		k.DeleteMarketMaker(ctx, mmAddr, p.PairId)
+
+		// refund DepositAmount
+		deposit, found := k.GetDeposit(ctx, mmAddr, p.PairId)
+		if !found {
+			return types.ErrInvalidDeposit
+		}
+		err = k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, mmAddr, deposit.Amount)
+		if err != nil {
+			return err
+		}
+		k.DeleteDeposit(ctx, mmAddr, p.PairId)
+
+		ctx.EventManager().EmitEvents(sdk.Events{
+			sdk.NewEvent(
+				types.EventTypeRejectMarketMaker,
+				sdk.NewAttribute(types.AttributeKeyAddress, p.Address),
+				sdk.NewAttribute(types.AttributeKeyPairId, fmt.Sprintf("%d", p.PairId)),
+			),
+		})
+	}
+	return nil
+}
+
+// DistributionMarketMakerIncentives is a handler for distribute incentives to eligible market makers.
+func (k Keeper) DistributionMarketMakerIncentives(ctx sdk.Context, proposals []types.IncentiveDistribution) error {
+	params := k.GetParams(ctx)
+	totalIncentives := sdk.Coins{}
+	for _, p := range proposals {
+		totalIncentives = totalIncentives.Add(p.Amount...)
+
+		mm, found := k.GetMarketMaker(ctx, p.GetAccAddress(), p.PairId)
+		if !found {
+			return types.ErrNotExistMarketMaker
+		}
+		if !mm.Eligible {
+			return types.ErrNotEligibleMarketMaker
+		}
+
+	}
```

Comment:
Remove the unnecessary space in between two brackets (Very minor stuff)
```suggestion
		}
	}
```